rasa/engine/runner/interface.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Text
+
+from rasa.engine.graph import ExecutionContext, GraphSchema
+
+
+class GraphRunner(ABC):
+    """A `GraphRunner` is responsible for running a `GraphSchema`."""
+
+    @classmethod
+    @abstractmethod
+    def create(
+        cls, graph_schema: GraphSchema, execution_context: ExecutionContext
+    ) -> GraphRunner:
+        """Creates a new instance of a GraphRunner.
+
+        Args:
+            graph_schema: The graph schema that will be instantiated and run.
+            execution_context: Context that will be passed to every `GraphComponent`.
+
+        Returns: Instantiated GraphRunner
+        """
+        ...
+
+    @abstractmethod
+    def run(
+        self,
+        inputs: Optional[Dict[Text, Any]] = None,
+        targets: Optional[List[Text]] = None,
```

Comment:
Most of the time, because each target node will specify it's a target. This is just for if we want to run different subsets of the graph. I think we added it because we thought we might just tokenize or something.

Reply:
Oh, I see. Makes sense 👍🏻